test.c

```diff
@@ -81,5 +72,55 @@ int main () {
 		else printf ("%12d |  %34s |  %34s  "MAGENTA"FAIL\n"RESET, arr[i], hexArr[i], binRes[i]);
 	}
 	printf ("-------------------------------------------------------------------------------------------------------\n\n");
+}
+
+///<summary>Gets decimal input from the user and converts it to binary and hexadecimal.</summary>
+void GetInput () {
+	int c;
+	while ((c = getchar ()) != '\n' && c != EOF);                                                 // Clear input buffer
+	int lenIn = 0;
+	char inStr[12];
+	printf ("Enter an integer : ");
+	while ((c = getchar ()) != '\n') {

```

Comment:
Try using fgets here. It gets the whole input as a string.
[fgets](https://www.geeksforgeeks.org/fgets-gets-c-language/)